src/index.js

```diff
@@ -23,62 +25,81 @@ export default class MagicUrl {
       if (typeof node.data !== 'string') {
         return
       }
-      const matches = node.data.match(this.options.globalRegularExpression)
-      if (matches && matches.length > 0) {
-        const newDelta = new Delta()
-        let str = node.data
-        matches.forEach(match => {
-          const split = str.split(match)
-          const beforeLink = split.shift()
-          newDelta.insert(beforeLink)
-          newDelta.insert(match, {link: this.normalize(match)})
-          str = split.join(match)
-        })
-        newDelta.insert(str)
-        delta.ops = newDelta.ops
-      }
+      const newDelta = new Delta()
+      node.data.split(/(\s+)/).forEach(str => {
+        const urlMatches = str.match(this.options.globalRegularExpression)
+        const mailMatches = str.match(this.options.globalMailRegularExpression)
+        if (urlMatches && urlMatches.length) {
+          urlMatches.forEach(match => {
+            const split = str.split(match)
+            const beforeLink = split.shift()
+            newDelta.insert(beforeLink)
+            newDelta.insert(match, {link: this.normalize(match)})
+            str = split.join(match)
+          })
+          newDelta.insert(str)
+        } else if (mailMatches && mailMatches.length) {
+          mailMatches.forEach(match => {
+            const split = str.split(match)
+            const beforeLink = split.shift()
+            newDelta.insert(beforeLink)
+            newDelta.insert(match, {link: `mailto:${match}`})
+            str = split.join(match)
+          })
+          newDelta.insert(str)
```

Comment:
This section could possible be more easy to read, if we abstract the match handling.

Something along those lines:

```js
function addMatchToDelta(match, normalizer) {
  const split = str.split(match)
  const beforeLink = split.shift()
  newDelta.insert(beforeLink)
  newDelta.insert(match, {link: normalizer(match)})
  str = split.join(match)
}

if (urlMatches && urlMatches.length) {
  urlMatches.forEach(match => addMatchToDelta(match, this.urlNormalizer))
} else if (mailMatches && mailMatches.length) {
  mailMatches.forEach(match => addMatchToDelta(match, this.mailNormalizer))
}
```

Reply:
Hi @visualjerk, thanks for your review! I've implmented the requested changes and updated the branch. Looking forward to get your feedback!

Reply:
Perfect 😃 Thanks again!